backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -163,4 +163,40 @@ const IR::IndexedVector<IR::DpdkAsmStatement> *RemoveLabelAfterLabel::removeLabe
     }
     return new_l;
 }
+
+const IR::Node* RemoveUnusedMetadataFields::preorder(IR::DpdkAsmProgram *p) {
+    auto usedFieldsCollector = new CollectUsedMetadataField();
+    usedFieldsCollector->setCalledBy(this);
+    p->apply(*usedFieldsCollector);
```

Comment:
you can probably call prune() too after this, this visitor does not need to traverse p.
This pattern of allocating the visitor in the preorder makes the program just a bit harder to read, why not just put CollectUsedMetadataFields in the list of backend passes and pass a reference to it to RemoveUnusedMetadataFields (or a reference to the used_fields structure)?
Frankly, it's not a big difference, it up to you, but the other pattern is the one used almost everywhere. We do allocate visitors locally too, but that's usually when we want to apply them to inner nodes.

Reply:
Incorporated the suggestions.